Fix use-after-move of playerView in VideoFullscreenManagerProxy::createViewWithID()

ddkilzer opened the pull request:
#### 809270197668234dc0c8eb099e11723eafa9c247
<pre>
Fix use-after-move of playerView in VideoFullscreenManagerProxy::createViewWithID()
<a href="https://bugs.webkit.org/show_bug.cgi?id=258093">https://bugs.webkit.org/show_bug.cgi?id=258093</a>
&lt;rdar://110799152&gt;

Reviewed by Chris Dumez.

Stop using WTFMove() on RetainPtr&lt;&gt; to fix the bug.

* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.mm:
(WebKit::VideoFullscreenManagerProxy::createLayerWithID):
- Whitespace change to match code structure in createViewWithID().
(WebKit::VideoFullscreenManagerProxy::createViewWithID):
- Stop using WTFMove(playerView) when calling setPlayerView() to fix the
  use-after-move.
- Make playerLayer a RetainPtr&lt;&gt; variable and use WTFMove(playerLayer)
  when calling setPlayerLayer() to make it match ARC lifetime and avoid
  ref count churn.

Canonical link: <a href="https://commits.webkit.org/265193@main">https://commits.webkit.org/265193@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cb6de9203f9845bda144d8c8521cc1612daede08

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10110 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10354 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10607 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11761 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9792 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12344 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10306 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12712 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10265 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11032 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8547 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12146 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8336 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16474 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9438 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9310 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12573 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9794 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7955 "2 flakes 3 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8953 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13191 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1146 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9597 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.mm:
```diff
@@ -708,7 +709,7 @@ - (BOOL)prefersStatusBarHidden
         auto playerView = adoptNS([allocWebAVPlayerLayerViewInstance() init]);
 
-        auto *playerLayer = (WebAVPlayerLayer *)[playerView layer];
+        auto playerLayer = RetainPtr { (WebAVPlayerLayer *)[playerView layer] };
```
I think we usually do:
```
RetainPtr playerLayer { (WebAVPlayerLayer *)[playerView layer] };
```

ddkilzer replied:
Fixed.  (Sorry, you keep flagging this on my patches. I will double-down on remembering to use this format.)

Commit:
Fix use-after-move of playerView in VideoFullscreenManagerProxy::createViewWithID()
https://bugs.webkit.org/show_bug.cgi?id=258093
<rdar://110799152>

Reviewed by Chris Dumez.

Stop using WTFMove() on RetainPtr<> to fix the bug.

* Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.mm:
(WebKit::VideoFullscreenManagerProxy::createLayerWithID):
- Whitespace change to match code structure in createViewWithID().
(WebKit::VideoFullscreenManagerProxy::createViewWithID):
- Stop using WTFMove(playerView) when calling setPlayerView() to fix the
  use-after-move.
- Make playerLayer a RetainPtr<> variable and use WTFMove(playerLayer)
  when calling setPlayerLayer() to make it match ARC lifetime and avoid
  ref count churn.

Canonical link: https://commits.webkit.org/265193@main
